component/loki/source/gelf/component.go

```diff
@@ -0,0 +1,141 @@
+package gelf
+
+import (
+	"context"
+	"sync"
+
+	"github.com/prometheus/common/model"
+
+	"github.com/grafana/agent/component"
+	"github.com/grafana/agent/component/common/loki"
+	"github.com/grafana/loki/clients/pkg/promtail/api"
+	"github.com/grafana/loki/clients/pkg/promtail/scrapeconfig"
+	"github.com/grafana/loki/clients/pkg/promtail/targets/gelf"
```

Comment:
In `loki.source.syslog` (and some other Loki components) we went with a slightly different approach; 
a) copying the targets package as an internal package for the component, and
b) making them both use our own copy of the `github.com/grafana/loki/clients/pkg/promtail/api` package in `github.com/grafana/agent/component/common/loki`.

It shouldn't make the component itself any harder to work with, just will slightly inflate the PR size but allow us to have a straightforward way to deprecate Promtail